cpp/src/slice2cs/Gen.cpp

```diff
@@ -1331,26 +1329,47 @@ Slice::Gen::TypesVisitor::TypesVisitor(IceUtilInternal::Output& out) :
 bool
 Slice::Gen::TypesVisitor::visitModuleStart(const ModulePtr& p)
 {
-    DictionaryList dicts;
-    if(p->hasOnlyDictionaries(dicts))
+    if (p->hasOnlyClassDecls() || p->hasOnlyInterfaces())
     {
-        //
-        // If this module contains only dictionaries, we don't need to generate
-        // anything for the dictionary types. The early return prevents
-        // an empty namespace from being emitted, the namespace will
-        // be emitted later by the dictionary helper .
-        //
-        return false;
+        return false; // avoid empty namespace
     }
 
     moduleStart(p);
     string name = fixId(p->name());
     _out << sp;
     emitCustomAttributes(p);
     _out << nl << "namespace " << name;
-
     _out << sb;
 
+    // Write constants if there is any
```

Comment:
Highly pedantic, but it should be "Write constants if there **_are_** any"

Reply:
Fixed.